### PR TITLE
Event Documentation Start

### PR DESCRIPTION
**Pull request by VapourDrive**
I began work on documenting events, with an overview, a general warning about the nature of events for novice users and a fleshed out example. With the start to finish registry example.

This is by no means a fully comprehensive covering of events, but a good introduction to them.
I am interested in providing more event documentation based on what is wanted.

### CODE REVIEW

File: events/usingevents.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+and
+'FMLCommonHandler.instance().bus().register(new YourEventClass());' for fml events
+You can use one class for all events that you wish to subscribe to, or spread them out over multiple.
+
+Your Event Use
+--------------
+
+The basics for subscribing to an event are as follows:
+
+'''
```
Comment (matthewprenger) [inline events/usingevents.md]: Wrong characters. You should be using the ` instead of ' for code blocks

**Comment by VapourDrive** [comment]
You commented just as I realized the misshap, sorry about that. fixing now.

**Comment by matthewprenger** [comment]
@VapourDrive Minor nitpick. You can use:
 \`\`\`java
    class Foo {}
 \`\`\`
to get code coloring.

**Comment by VapourDrive** [comment]
That does look a lot nicer! I'll remember that for next time.

**Comment by cpw** [comment]
Can you please correct some spelling mistakes? Wether is whether throughout the document. Sorry, I just want to make sure the little things don't bug me..

**Comment by cpw** [comment]
@VapourDrive why the sudden close attempt? This looks ok, but it needs a little bit of tidying up is all..

**Comment by VapourDrive** [comment]
I can do make the spelling mistake changes, I'll scan the document for any other issues.

**Comment by VapourDrive** [comment]
I have fixed the two instances of "wether" and added the paths to the the mkdocs file. 
@cpw can you confirm that the mkdocs file is what is desired? And maybe provide feedback on the content and style.

File: events/overview.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Overview
+========
+
+The event system in forge/fml provides a way for a mod to change something that happens in the game without breaking down compatibility.
+
+An event is void function that gets fired when "an event" happens in the game and gives mods the opportunity to change the outcome by subscribing to the event.
```
Comment (cpw) [inline events/overview.md]: Is a void function?

File: events/usingevents.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+{
+	@SubscribeEvent(priority = EventPriority.LOWEST)
+	public void WhateverYouChoseForFunctioName(NameOfEvent event)
+	{
+		//your code
+	}
+}
+```
+
+The event priority that is **optional** to set determines what order the subscribers get acces to the event
```
Comment (cpw) [inline events/usingevents.md]: access

File: events/usingevents.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+If you have the lowest priority you get access last, but you get the last chance to set what happens in the event.
```
Comment (cpw) [inline events/usingevents.md]: called, not accessed. The priority affects the calling sequence for dispatching events to the handlers.

File: events/usingevents.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	{
+		if(event.entityPlayer.experienceLevel > 0)
+		{
+			event.newSpeed = event.originalSpeed + 1.0F;
+		}
+	}
+}
+```
+
+The newSpeed float is the only variable that can actually be set set, as all the others are final. Here we simply add one float to the original speed if the player has greater than 0 levels .
```
Comment (cpw) [inline events/usingevents.md]: In this example, the newSpeed float is the only factor that can be changed. In this case, we are adding one to it, increasing the break speed, for an experienced player.

File: events/usingevents.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+The event can be canceled.
```
Comment (cpw) [inline events/usingevents.md]: This specific event can be cancelled, causing the break to not start.

File: events/eventnature.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+EventNature
+===========
+
+When you want to do something with an event, you register your subscription, and you set up your block of code that gets run.
+(More on the specifics elsewhere)
```
Comment (cpw) [inline events/eventnature.md]: Where?

File: events/usingevents.md (hunk 4 above)
Comment (VapourDrive) [inline events/usingevents.md]: Fair enough. Proper terminology is key for good documentation.

File: events/usingevents.md (hunk 5 above)
Comment (VapourDrive) [inline events/usingevents.md]: This is probably a more eloquent, and less ambiguous way of describing what is happening.

File: events/eventnature.md (hunk 7 above)
Comment (VapourDrive) [inline events/eventnature.md]: I'll reference the example that I made.

**Comment by VapourDrive** [comment]
I'll try to change what needs changing as soon as possible.

**Comment by LexManos** [comment]
Poke for update.

**Pull request closed without merging**